util/src/main/java/org/datacommons/util/ComplexValueParser.java

```diff
@@ -0,0 +1,335 @@
+    arg.delimiter = ' ';
+    arg.includeEmpty = false;
+    arg.stripEnclosingQuotes = false;
+    // TODO: Passthru errCb
+    List<String> fields = splitAndStripWithQuoteEscape(trimmedComplexValue, arg, null);
+    if (fields.isEmpty()) {
+      logCtx.addEntry(
+          Debug.Log.Level.LEVEL_ERROR,
+          "MCF_MalformedComplexValueString",
+          "Found malformed complex value '"
+              + complexValue
+              + "' in property "
+              + prop
+              + " in "
+              + "node "
+              + mainNodeId,
+          mainNode.getLocationsList());
+      return false;
+    }
+    if (fields.size() != 2 && fields.size() != 3) {
+      logCtx.addEntry(
+          Debug.Log.Level.LEVEL_ERROR,
+          "MCF_MalformedComplexValueParts",
+          "Complex value must have 2 or 3 components but '"
+              + complexValue
+              + "' in property "
+              + prop
+              + " in "
+              + "node "
+              + mainNodeId
+              + " has "
+              + fields.size(),
+          mainNode.getLocationsList());
+      return false;
+    }
+    boolean isRange = fields.size() == 3;
+
+    // Find out the offset of the various fields.
+    int startIdx = -1, endIdx = -1, valueIdx = -1, unitIdx = -1;
+    if (fields.get(0).charAt(0) == '-' || Character.isDigit(fields.get(0).charAt(0))) {
+      // First field is value
+      if (isRange) {
+        unitIdx = 2;
+        startIdx = 0;
+        endIdx = 1;
+      } else {
+        unitIdx = 1;
+        valueIdx = 0;
+      }
+    } else {
+      // First field is unit
+      if (isRange) {
+        unitIdx = 0;
+        startIdx = 1;
+        endIdx = 2;
+      } else {
+        unitIdx = 0;
+        valueIdx = 1;
+      }
+    }
+
+    // Get unit.
+    String unit;
+    int colonIndex = fields.get(unitIdx).indexOf(Vocabulary.REFERENCE_DELIMITER);
+    if (colonIndex != -1) {
+      unit = fields.get(unitIdx).substring(colonIndex + 1);
+    } else {
+      unit = fields.get(unitIdx);
+    }
+
+    // Compute DCID.
+    boolean isLatLng = false;
+    if (fields.size() == 2) {
+      if (!McfUtil.isNumber(fields.get(valueIdx))) {
+        logCtx.addEntry(
+            Debug.Log.Level.LEVEL_ERROR,
+            "MCF_QuantityMalformedValue",
+            "Quantity value '"
+                + complexValue
+                + "' in property "
+                + prop
+                + " in node "
+                + mainNodeId
+                + " must be a number",
+            mainNode.getLocationsList());
+        return false;
+      }
+      dcid = unit + fields.get(valueIdx);
+      name = unit + " " + fields.get(valueIdx);
+    } else {
+      assert fields.size() == 3;
+      if (unit.toLowerCase().equals(Vocabulary.LAT_AND_LONG.toLowerCase())) {
+        isLatLng = true;
+        if (!parseLatLng(fields.get(startIdx), fields.get(endIdx))) {
+          return false;
```

Comment:
maybe add a note here that the errors are generated in the follow on parseLatLng function?

Reply:
Done.